Friendly-plans/app/src/main/java/pg/autyzm/friendly_plans/TaskContainerFragment.java

```diff
@@ -25,30 +34,51 @@
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
             Bundle savedInstanceState) {
-        ((App) getActivity().getApplication()).getRepositoryComponent().inject(this);
+        ((App) getActivity().getApplication()).getAppComponent().inject(this);
         return inflater.inflate(R.layout.fragment_task_container, container, false);
     }
 
     @Override
     public void onViewCreated(View view, Bundle savedInstanceState) {
+        registerViews(view);
+        taskNext.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+            if (taskValidation.isValid(taskName, taskDurTime)) {
+                addTaskOnDb();
+                goToNextPage();
+            }
+            }
+        });
+    }
+
+    private void addTaskOnDb() { //TODO:  how about exception handling of db transaction ?
```

Comment:
Please remove TODO comments, if something is needed, please create issues for it.